code-challenges.test.js

```diff
@@ -14,39 +14,135 @@
+    // Object Destructuring. Use Dot notation for the value of name. Use Built-in Method .split(" "), use space as a delimiter.
+    // Built-in Method will return an array with a length of two. 
+    // Variable firstName is set to the first element in the array. Variable lastName is set to the second element in the array.
+    [firstName, lastName] = value.name.split(" ")
+  // Delcare new variable. Call upon our destructured variable firstName.
+  // Use Built-in Methods .charAt(), .toUpperCase() to indicate the first index.
+  // Use arithmetic Operator +, and Built-in Method .substring(1) to extract characters from our starting index of (1). Returns a substring.
+  const firstNameUpperCase = firstName.charAt(0).toUpperCase() + firstName.substring(1)
+  // Delcare new variable. Call upon our destructured variable lastName.
+  // Use Built-in Methods .charAt(), .toUpperCase() to indicate the first index.
+  // Use arithmetic Operator +, and Built-in Method .substring(1) to extract characters from our starting index of (1). Returns a substring.
+  const lastNameUpperCase = lastName.charAt(0).toUpperCase() + lastName.substring(1)
+  // Declare a new variable. Use dot notation to call upon the occupation.
+  const addOcuppation = value.occupation
+  // Return a String Interpolation with firstNameUpperCase, lastNameUpperCase, and addOcuppation. Add "is" and "." to match expected outcome.
+  return `${firstNameUpperCase} ${lastNameUpperCase} is ${addOcuppation}.`
+  })
+  // Return firstLastNameCapital outside of the HOF .map.
+  return firstLastNameCapital
+}
+
+
 // Pseudo code:
+// Create a function called sentenceWithFirstLastCapitalized. Takes one parameter called arrayOfObjects.
+// Declare a new variable named firstLastNameCapital
+// Use HOF .map to iterate on our arrayOfObjects with one parameter called value.
+// We call upon object destructuring with variable names firstName and lastName. Since it is a object use dot notation to call upon .name, use built in method .split(" "). Thus will specify our first element using .split(" ") to firstName and our second element to lastName.
+// Declare a new variable named firstNameUpperCase. 
+    // Call upon firstName with built in methods .charAt() and .toUpperCase() * can use these because it is a string
+    // Use arithmetic operator and built in method .substring(1) to return a substring of the original string.
+// declare a new variable named lastNameUpperCase and do the same thing but change firstName to lastName.
+// declare a new variable called addOcuppation.
+    // Since it is an object we use dot notation to pull the value from the key .occupation.
+// return a string interpolation for firstNameUpperCase, lastNameUpperCase and addOccupation. Look at the expected output and insert "is" and a "." within the string interpolation.
+// return firstLastNameCapital outside of the HOF block but within setenceWithFirstLastCapitalized function.
```

Comment:
Great logic. I especially liked the use of destructuring, and setting up the firstName and lastName variables, good work! Also, you have in line pseudo code, so no need to add this pseudo code here again